echopype/convert/set_groups_base.py

```diff
@@ -362,3 +362,141 @@ def beam_groups_to_convention(
 
         self._add_ping_time_dim(ds, beam_ping_time_names, ping_time_only_names)
         self._add_beam_dim(ds, beam_only_names, beam_ping_time_names)
+
+    def _add_index_data_to_platform_ds(
+        self,
+        platform_ds: xr.Dataset,
+    ) -> xr.Dataset:
+        """
+        Append index data from `.IDX` file to the `Platform` dataset.
+        Index file data contains latitude, longitude, and vessel distance traveled.
+
+        Parameters
+        ----------
+        platform_ds : xr.Dataset
+            `Platform` dataset without `.IDX` data.
+
+        Returns
+        -------
+        platform_ds : xr.Dataset
+            `Platform` dataset with `.IDX` data.
+            Contains new `time3` dimension to correspond with `.IDX` timestamps that
+            align with `vessel_distance`, `latitude_idx`, and `longitude_idx`.
+
+        Notes
+        -----
+        This function is only called for EK60/EK80 conversion.
+        """
+        timestamp_array, _, _ = xr.coding.times.encode_cf_datetime(
+            np.array(self.parser_obj.idx["timestamp"]),
+            **{
+                "units": DEFAULT_TIME_ENCODING["units"],
+                "calendar": DEFAULT_TIME_ENCODING["calendar"],
+            },
+        )
+        platform_ds = platform_ds.assign(
+            {
+                "vessel_distance": xr.DataArray(
+                    np.array(self.parser_obj.idx["vessel_distance"]),
+                    dims=("time3"),
+                    coords={"time3": timestamp_array},
+                ),
+                "latitude_idx": xr.DataArray(
+                    np.array(self.parser_obj.idx["latitude_idx"]),
+                    dims=("time3"),
+                    coords={"time3": timestamp_array},
+                ),
+                "longitude_idx": xr.DataArray(
+                    np.array(self.parser_obj.idx["longitude_idx"]),
+                    dims=("time3"),
+                    coords={"time3": timestamp_array},
+                ),
+            }
+        )
+        platform_ds["time3"] = platform_ds["time3"].assign_attrs(
+            {
+                "axis": "T",
+                "long_name": "Timestamps from `.IDX` datagrams",
```

Comment:
Same minor thing: remove all backticks from the comment here and below?
```suggestion
                "long_name": "Timestamps from .IDX datagrams",
```